### PR TITLE
Feature popup conditions

### PR DESCRIPTION
**Pull request by crazypsycho**
Add conditions for popups and refactor to new standards

### CODE REVIEW

File: trunk/Public/DynamicConditionsPublic.php [hunk 1]
```diff
@@ -235,6 +239,26 @@ private function convertAcfDate( $id, array $data ) {
         $this->elementSettings[$id]['dynamicconditions_dynamic'] = $timestamp;
     }
 
+
+    /**
+     * Hooks into elementor/theme/before_do_popup, loops all popups and check for condition
+     *
+     * @param Locations_Manager $locationManager
+     */
+    public function beforePopup( $locationManager ) {
```
[the remark this answers is not in the document]
Comment (crazypsycho) [reply trunk/Public/DynamicConditionsPublic.php]: I don´t see a real need for this, cause the methods are not made to be called outside the plugin.
But i made a Legacy-Date-Class.

File: trunk/Public/DynamicConditionsPublic.php [hunk 2]
```diff
@@ -235,6 +239,26 @@ private function convertAcfDate( $id, array $data ) {
         $this->elementSettings[$id]['dynamicconditions_dynamic'] = $timestamp;
     }
 
+
+    /**
+     * Hooks into elementor/theme/before_do_popup, loops all popups and check for condition
```
[the remark this answers is not in the document]
Comment (crazypsycho) [reply trunk/Public/DynamicConditionsPublic.php]: Should be better now

[the remark this answers is not in the document]
Comment (eac0n) [reply trunk/Public/DynamicConditionsPublic.php]: Please remove this line.  
A method does not determine when or by whom it is called. Furthermore the statement about the internal implementation is not relevant.

File: trunk/Public/DynamicConditionsPublic.php [hunk 3]
```diff
@@ -235,13 +233,34 @@ private function convertAcfDate( $id, array $data ) {
...
+     * Removes popup from location, if it is hidden
```
Comment (eac0n) [inline trunk/Public/DynamicConditionsPublic.php]: Does this method "check popups condition" or "remove a/the popup from a/the/it's location if it (the popup or the location?) is hidden"?  
The description and the method name do not really match.

File: trunk/Public/DynamicConditionsPublic.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (crazypsycho) [reply trunk/Public/DynamicConditionsPublic.php]: The method is only called by the hook. So i think this is relevant, You can see directly that it used before_do_popup.
But to make you happy, i have removed it :-)

File: trunk/Lib/Date.php [hunk 4]
```diff
@@ -131,10 +129,12 @@ private static function getMonths() {
...
      */
     public static function getDaysTranslated() {
-        $currentLocale = setlocale( LC_ALL, 0 );
-        setlocale( LC_ALL, get_locale() );
         $dayList = self::getDays();
-        setlocale( LC_ALL, $currentLocale );
+
+        // translate by wordpress-lang
+        foreach ( $dayList as &$day ) {
+            $day = __( $day );
```
Comment (eac0n) [inline trunk/Lib/Date.php]: same as line 106

File: trunk/Lib/Date.php [hunk 5]
```diff
@@ -103,10 +99,12 @@ public static function unTranslateDate( $needle = '', $setLocale = null ) {
...
      */
     public static function getMonthsTranslated() {
-        $currentLocale = setlocale( LC_ALL, 0 );
-        setlocale( LC_ALL, get_locale() );
         $monthList = self::getMonths();
-        setlocale( LC_ALL, $currentLocale );
+
+        // translate monthlist by wordpress-lang
+        foreach ( $monthList as &$month ) {
+            $month = __( $month );
```
[the remark this answers is not in the document]
Comment (crazypsycho) [reply trunk/Lib/Date.php]: I think this is inevitable on this point.

File: trunk/Lib/Date.php (hunk 4 above)
Comment (crazypsycho) [reply trunk/Lib/Date.php]: I think this is inevitable on this point.

File: trunk/Lib/Date.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (crazypsycho) [reply trunk/Lib/Date.php]: It is now hardcoded

File: trunk/Lib/Date.php (hunk 4 above)
Comment (crazypsycho) [reply trunk/Lib/Date.php]: It is now hardcoded

**Pull request merged**